test/prim/pir_prim/test_auto_recompute.py

```diff
@@ -153,11 +153,11 @@ def test_auto_recompute(self):
                 atol=TOLERANCE[self.dtype]["atol"],
                 rtol=TOLERANCE[self.dtype]["rtol"],
             )
-            forward_ops = recompute_program.global_block().ops[:14]
-            backward_ops = recompute_program.global_block().ops[14:]
-            saved_values = forward_ops[9].results()[0]
+            forward_ops = recompute_program.global_block().ops[:13]
+            backward_ops = recompute_program.global_block().ops[13:]
+            saved_values = forward_ops[10].results()[0]
```

Comment:
Is this kind of hard coding of numbers making maintenance difficult, and are there any better methods?

Reply:
This op is the dividing point in recompute strategy, and it is hard to determine in advance manual.

Reply:
TODO: add interface to maintenance forward op end index information in to static.